### ISSUE TITLE
Question on the page "instantsearch"

### ISSUE THREAD

**Issue by daveiglu** [algolia/instantsearch.js#2985]
Hi! I have a documentation issue on InstantSearch.js. I could not find what I was looking for on the page "[instantsearch](https://community.algolia.com/instantsearch.js/v2/instantsearch.html

I am trying to automatically re-run the search only if the first attempt produces no results - trying to remove some applied refinements.
I understand that the only event raised from the search is "render" but it obviously runs only after all the widget have been rendered.
Is there any event before render and just after the search (e.g. search completed) that I can handle and in which I can check the results and possibly change the refinements (and re-run the search again)?
If not, how can I manage this scenario?

Thanks

**Comment by bobylito** [comment]
Hi @daveiglu 

Indeed from the InstantSearch perspective there is only the `render` event. Underneath, however there are other events managed by the [JS Helper](https://community.algolia.com/algoliasearch-helper-js/). For example, there is [an event where new results](https://community.algolia.com/algoliasearch-helper-js/reference.html#AlgoliaSearchHelper#event:result) are received:

```js
// after search.start() has been called
search.helper.on('result', (results, state) => {
  // results contains the actual results
  // state contains the state used to generate those results
});
```

**Comment by Haroenv** [comment]
Perfect, feel free to open a new issue if you have any other questions!
